test/contracts/common/erc20-transfer.ts

```diff
@@ -0,0 +1,79 @@
+import {ExpectTokenTransfer} from '../bond/single-collateral-bond-events'
+import {ActualTokenTransfer} from '../bond/verify-single-collateral-bond-events'
+import {BigNumber, Event} from 'ethers'
+import {TransferEvent} from '../../../typechain-types/IERC20'
+import {expect} from 'chai'
+import {Result} from '@ethersproject/abi'
+
+export function deepEqualsERC20TokenTransfer(
+    actual: ActualTokenTransfer,
+    expected: ExpectTokenTransfer
+): boolean {
+    return (
+        actual.to === expected.to &&
+        actual.from === expected.from &&
+        actual.value.toBigInt() === expected.amount
+    )
+}
+
+/**
+ * Shape check and conversion for a TransferEvents.
+ */
+export function erc20TransferEvents(events: Event[]): {
+    from: string
+    to: string
+    value: BigNumber
+}[] {
+    const converted: {
+        from: string
+        to: string
+        value: BigNumber
+    }[] = []
+
+    for (let i = 0; i < events.length; i++) {
+        const transfer = events[i] as TransferEvent
+        expect(transfer.args).is.not.undefined
+
+        const args = events[i].args
+        expect(args?.from).is.not.undefined
+        expect(args?.to).is.not.undefined
+        expect(args?.value).is.not.undefined
+
+        converted.push(transfer.args)
+    }
+
+    return converted
+}
+export type ExpectedERC20Transfer = {
+    to: string
+    from: string
+    amount: bigint
+}
+
+export function erc20TransferEventLogsFromResult(
+    events: Result
+): ExpectedERC20Transfer[] {
```

Comment:
question: should this be return `ActualERC20Transfer[]`, as the `Expected` prefix is generally used in testing for what you expect in equality or verify evaluations.

This is why you'll see some of the classes with both `Actual` and `Expected` types that have the same shape.

Reply:
should be ActualERC20Transfer imo. I've changed the return type and also moved the type definition to this file because that makes more sense?

Reply:
Are both `ActualERC20Transfer` and `ConvertedTransferEvent` now required?

Reply:
Related: Why are we mixing both bigint and BigNumber for the amount fields of the transfer events.

Reply:
How do you mean?